package/MDAnalysis/analysis/rms.py

```diff
@@ -335,6 +335,19 @@ class RMSD(AnalysisBase):
        :attr:`RMSD.rmsd` instead.
 
     """
+    def __weight_type_check(self, weights, atoms, selection):
+        if (np.array(weights).ndim == 1) and (np.array(weights).dtype 
+                                in (np.dtype('float64'),np.dtype('int64'))):
+            if len(weights) != atoms.n_atoms:
+                raise ValueError("Length of provided weights {} do not match the number of atoms "
+                      "in the selection {}: {}".format(weights, selection['mobile'], atoms.n_atoms))
+            else:
+                get_weights(atoms, weights) 
+        elif not iterable(weights) and str(weights) != 'mass' and weights is not None:
+            raise ValueError("Each groupselection can only be combined with "
+                           "weights: None, 'mass' or 1D float array")
```

Comment:
So having now looked at `get_weights`, it looks like that already does most of these checks/raises errors, so might be easier to use that rather than adding a new function. It would be useful to keep the more specific error message identifying which selection cause the error, since this wouldn't be clear from `get_weights` alone, but you could use say try/except to catch and expand on the errors when calling `get_weights`

Reply:
Now the error raises as:
````
u = mda.Universe(PSF, DCD)
R = rms.RMSD(u, 
             u,              
             groupselections=['name CA and resid 1-5','name CA and resid 1'],
             weights_groupselections=[[1,1,1,1,1],[1,1]]) 

R.run()

weights (length 2) must be of same length as the atoms (1) happens in selection ['name CA and resid 1']
````